doc/man3/SSL_set_retry_verify.pod

```diff
@@ -0,0 +1,67 @@
+=pod
+
+=head1 NAME
+
+SSL_set_retry_verify - indicate that certificate verification should be retried
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ int SSL_set_retry_verify(SSL *ssl);
+
+=head1 DESCRIPTION
+
+SSL_set_retry_verify() should be called from the certificate verification
+callback when the application wants to indicate that the handshake should be
+suspended and the control should be returned to the application. The application
+should then call SSL_connect() again to retry the verification callback and
+to resume the handshake.
+
+=head1 NOTES
+
+The effect of calling SSL_set_retry_verify() outside of the certificate
+verification callback is undefined.
```

Comment:
Good point, retries should ideally also be available when servers are verifying client certificates, with `SSL_accept(3)` called again after updating the certificate store (or otherwise gathering additional certificates) to make verification possible.

Perhaps also a context grep for `SSL_ERROR_WANT_RETRY_VERIFY` is warranted to ensure that all relevant documentation is updated, and all code paths are correct.

Reply:
> Good point, retries should ideally also be available when servers are verifying client certificates, with `SSL_accept(3)` called again after updating the certificate store (or otherwise gathering additional certificates) to make verification possible.

That would be a new feature which we cannot backport to 3.0.

Reply:
So I'd reserve handling server side for another PR.

Reply:
> So I'd reserve handling server side for another PR.

While adding server-side support is outside this PR scope, _documenting_ what happens if the retry feature is (incorrectly) used on the server side can be considered in this PR scope. Such documentation may even help with smooth introduction of server-side support in the future. The documentation can just claim "undefined behavior" if that is the best we can do right now. Documenting UB is still better than saying nothing IMHO.

Reply:
> The documentation can just claim "undefined behavior" if that is the best we can do right now. Documenting UB is still better than saying nothing IMHO.

I did exactly that.